Entity/Mail.php

```diff
@@ -408,6 +425,37 @@ public function setStatus($status)
         return $this;
     }
 
+    /**
+     * get Headers.
+     *
+     * @return array
+     */
+    public function getHeaders()
+    {
+        return $this->headers;
+    }
+
+    /**
+     * set Headers.
+     *
+     * @param HeaderSet $headers The headers of the email
+     *
+     * @return Mail
+     */
+    public function setHeaders($headers)
+    {
+        $headerNames = $headers->listAll();
+        $headersArray = [];
+        foreach($headerNames as $headerName) {
+          if (!$this->isDefaultHeader($headerName) && $headers->get($headerName) instanceof \Swift_Mime_Headers_UnstructuredHeader) {
```

Comment:
Can you tell me why you're checking both ? I guess a simple `Swift_Mime_Headers_UnstructuredHeader` type check would be sufficient, isn't it ?